src/imperative/cached_op.cc

```diff
@@ -823,9 +823,10 @@ OpStatePtr CachedOp::DynamicForward(
 
   // If we are already recording, we don't need RunGraph to record all
   // computation again.
```

Comment:
I recommend clarifying in the comment that "When CachedOp is inlining, RunGraph is in charge of recording the graph, otherwise we are in charge of recording and RunGraph shall not be recording".